org.eclipse.xtext/src/org/eclipse/xtext/mwe/PathTraverser.java

```diff
@@ -54,20 +55,10 @@ public Set<URI> findAllResourceUris(String path, Predicate<URI> isValidPredicate
 
 	protected Set<URI> traverseArchive(File file, Predicate<URI> isValidPredicate) {
 		try {
-			Set<URI> result = Sets.newHashSet();
-			ZipFile zipFile = new ZipFile(file);
-			try {
-				Enumeration<? extends ZipEntry> entries = zipFile.entries();
-				while (entries.hasMoreElements()) {
-					ZipEntry entry = entries.nextElement();
-					URI uri = getUri(file, entry);
-					if (uri != null && isValidPredicate.apply(uri)) {
-						result.add(uri);
-					}
-				}
-				return result;
-			} finally {
-				zipFile.close();
+			try (ZipFile zipFile = new ZipFile(file);) {
+				return zipFile.stream().map(entry -> getUri(file, entry)) //
+						.filter(Objects::nonNull).filter(isValidPredicate) //
+						.collect(Collectors.toSet());
```

Comment:
Another subtle change in behavior. `Collectors.toSet` doesn't make any API guarantees about mutability of the returned set. Regardless of my personal preference of iteration vs stream pipelines, I'd go for `collect(Collectors.toCollection(HashSet::new))`

Reply:
That's right, but since `PathTraverser.traverseArchive()` respectively `findAllResourceUris()` does not make any guarantees as well I assumed it's fine to change this implementation detail. However, I applied your suggestion.